simphony/cuds/material.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from simphony.core.data_container import DataContainer
+
+
+class Material(object):
+    """A class that represents a material.
+
+    Class describes a material and its data.
+
+    Attributes
+    ----------
+    description : str
+        textual description of the material
+    uid : uuid.UUID
+        the id of the material. If None, then it is generated
+    data : DataContainer
+        data associated with this material
+
+    """
+    def __init__(self, uid=None, data=None, description=""):
```

Comment:
> You can surely add a separate notion of name with all the required machinery

Yeah, we should be able to query by name of by uid. There are at least three options to make it happen:
1. Write separate ```get_material_by_uid(uid)``` and ```get_material_by_name(name)``` methods
2. Write one method with two optional arguments ```get_material(name=None, uid=None)```
3. Write one method with one argument, and add a check inside the method, if this argument is of type ```str``` or ```uuid```, e.g.
```
class Materials(object):
     ...
     def get_material(id):
          if isinstance(id, uuid):
              ...
          elif isinstance(id, str):
              ...
```           
I prefer option 3. for it's simplicity.